evap/staff/importers/user.py

```diff
@@ -0,0 +1,383 @@
+
+        self.tracker: FirstLocationAndCountTracker = FirstLocationAndCountTracker()
+
+    def check_userdata(self, user_data: UserData, location: ExcelFileLocation):
+        if user_data not in self.first_location_by_user_data:
+            self.first_location_by_user_data[user_data] = location
+            return
+
+        first_location = self.first_location_by_user_data[user_data]
+        self.tracker.add_location_for_key(location=location, key=first_location)
+
+    def finalize(self) -> None:
+        for first_location, location_string in self.tracker.aggregated_keys_and_location_strings():
+            self.importer_log.add_warning(
+                _("{location}: The duplicated row was ignored. It was first found at {first_location}.").format(
+                    location=location_string,
+                    first_location=first_location,
+                ),
+                category=ImporterLogEntry.Category.IGNORED,
+            )
+
+
+class UserDataAdapter(RowCheckerMixin):
+    """Adapter to use Checkers for UserData with UserParsedRow"""
+
+    def __init__(self, user_data_checker):
+        self.user_data_checker = user_data_checker
+
+    def check_row(self, row: UserParsedRow):
+        self.user_data_checker.check_userdata(row.user_data, row.location)
+
+    def finalize(self) -> None:
+        self.user_data_checker.finalize()
+
+
+@transaction.atomic
+def import_users(excel_content, test_run: bool) -> Tuple[Iterable[UserProfile], ImporterLog]:
+    importer_log = ImporterLog()
+
+    with ConvertExceptionsToMessages(importer_log):
+        excel_mapper = ExcelFileRowMapper(skip_first_n_rows=1, row_cls=UserInputRow, importer_log=importer_log)
+        raw_rows = excel_mapper.map(excel_content)
+        importer_log.raise_if_has_errors()
+
+        rows = [raw_row.as_parsed_row() for raw_row in raw_rows]
+
+        for checker in [
+            UserDataAdapter(UserDataMismatchChecker(test_run, importer_log)),
+            UserDataAdapter(DuplicateUserDataChecker(test_run, importer_log)),
+            UserDataAdapter(UserDataEmptyFieldsChecker(test_run, importer_log)),
+            UserDataAdapter(UserDataValidationChecker(test_run, importer_log)),
+        ]:
+            checker.check_rows(rows)
+        importer_log.raise_if_has_errors()
+
+        users = [row.user_data for row in rows]
+
+        # Both will contain the updated data from the UserData instances, but the existing users are not yet saved.
+        existing_user_profiles, new_user_profiles = get_user_profile_objects(users)
+        resulting_user_profiles = existing_user_profiles + new_user_profiles
+
+        importer_log.raise_if_has_errors()
+        if test_run:
+            importer_log.add_success(_("The test run showed no errors. No data was imported yet."))
+            msg = format_html(_("The import run will create {} users:"), len(new_user_profiles))
+            msg += create_user_list_html_string_for_message(new_user_profiles)
```

Comment:
How do you feel about:
```suggestion
            msg = format_html(
                _("The import run will create {} users:{}"),
                len(new_user_profiles),
                create_user_list_html_string_for_message(new_user_profiles),
            )
```
Its slightly longer, but doesn't involve mutation of `msg` (same for the `else` case)